progressist/__init__.py

```diff
@@ -115,12 +115,28 @@ def speed(self):
         """Number of iterations per second."""
         return Float(1.0 / self.avg if self.avg else 0)
 
+    @property
+    def throttled(self):
+        if not self.throttle:
+            return
+        if isinstance(self.throttle, (int, float)):
+            throttle = self.throttle
+            if isinstance(self.throttle, float):
+                throttle = max(1, self.total * self.throttle)
+            throttle = self._last_render + throttle
+            if self.done < throttle:
+                if not self.total or throttle <= self.total:
+                    return True
+            self._last_render = self.done
+        elif isinstance(self.throttle, datetime.timedelta):
+            if ((not self.total or self.done < self.total) and
+               self._last_render + self.throttle.seconds > time.time()):
+                return True
+            self._last_render = time.time()
+
```

Comment:
No `else` ? So if `self.throttle` is anything else than an int, a float or a datetime then it's not throttled at all?

Just like for the float > 1.0, it seems to me like you should validate that value of `throttle` in the constructor maybe?